src/us/kbase/workspace/test/kbase/JSONRPCLayerTester.java

```diff
@@ -573,16 +573,26 @@ public boolean equals(Object obj) {
 	}
 	
 	@SuppressWarnings("deprecation")
-	protected void checkProvenance(String user, ObjectIdentity id,
-			List<ProvenanceAction> prov, Map<String, String> refmap,
-			Map<StringEpoch, StringEpoch> timemap) throws Exception {
+	protected void checkProvenance(
+			final String user,
+			final ObjectIdentity id,
+			final List<ProvenanceAction> prov,
+			final Map<String, String> refmap,
+			final Map<StringEpoch, StringEpoch> timemap)
+			throws Exception {
 		Date tenback = getOlderDate(10 * 60 * 1000);
 		Date tenfor = getNewerDate(10 * 60 * 1000);
 		
 		//get objs 2 prov
-		ObjectData ret1p = CLIENT1.getObjects2(new GetObjects2Params()
-			.withNoData(1L)
-			.withObjects(Arrays.asList(toObjSpec(id)))).getData().get(0);
+		final ObjectData ret1p;
+		try {
+			ret1p = CLIENT1.getObjects2(new GetObjects2Params()
+					.withNoData(1L)
+					.withObjects(Arrays.asList(toObjSpec(id)))).getData().get(0);
+		} catch (ServerException e) {
+			System.err.println(e.getData());
+			throw e;
```

Comment:
Will this throw and abort the rest of the tests in this function? Is there a reason why this one has the try/catch? If this `getObjects2` fails, will all the following tests/invocations of `getObjects2` also fail?

Reply:
> Will this throw and abort the rest of the tests in this function?

yes

> Is there a reason why this one has the try/catch?

The `toString` method of `ServerException` doesn't include the `data` field (and def shouldn't), which is the server side stack trace and so makes debugging the issue much easier. This particular call was failing server side so I added the try catch to get the server side stack trace. Since it doesn't hurt anything to leave it there, I did that.

> If this getObjects2 fails, will all the following tests/invocations of getObjects2 also fail?

Isn't this the same as the first question? If not can you elaborate as I'm a bit fogged re the difference

Reply:
It's the same Q phrased very slightly differently to befuddle and confuse.